ddtrace/internal/remoteconfig/products/apm_tracing.py

```diff
@@ -0,0 +1,83 @@
+from ddtrace import config
+from ddtrace.internal.core.event_hub import dispatch
+from ddtrace.internal.core.event_hub import on
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.remoteconfig._connectors import PublisherSubscriberConnector
+from ddtrace.internal.remoteconfig._publishers import RemoteConfigPublisher
+from ddtrace.internal.remoteconfig._pubsub import PubSub
+from ddtrace.internal.remoteconfig._subscribers import RemoteConfigSubscriber
+
+
+requires = ["remote-configuration"]
+
+
+log = get_logger(__name__)
+
+
+def _rc_callback(data, test_tracer=None):
+    for metadata, data in zip(data["metadata"], data["config"]):
+        if metadata is None or not isinstance(data, dict):
+            continue
+
+        service_target = data.get("service_target")
+        if service_target is not None:
+            service = service_target.get("service")
+            if service is not None and service != config.service:
+                continue
+
+            env = service_target.get("env")
+            if env is not None and env != config.env:
+                continue
+
+        lib_config = data.get("lib_config")
+        if lib_config is not None:
+            dispatch("apm-tracing.rc", (lib_config,))
+
+
+class APMTracingAdapter(PubSub):
+    __publisher_class__ = RemoteConfigPublisher
+    __subscriber_class__ = RemoteConfigSubscriber
+    __shared_data__ = PublisherSubscriberConnector()
+
+    def __init__(self):
+        self._publisher = self.__publisher_class__(self.__shared_data__)
+        self._subscriber = self.__subscriber_class__(self.__shared_data__, _rc_callback, "APM_TRACING")
+
+
+def post_preload():
+    pass
+
+
+def start():
+    if config._remote_config_enabled:
+        from ddtrace.internal.products import manager
+        from ddtrace.internal.remoteconfig.worker import remoteconfig_poller
+
+        remoteconfig_poller.register(
```

Comment:
We are registering the `APM_TRACING` product twice (once here and once in [ddtrace.settings._core](https://github.com/DataDog/dd-trace-py/blob/8bedc18c981af0414424fa350fd030a3d0d2ce10/ddtrace/settings/_config.py#L925-L926)). Can we consolidate these components?

Also it would be nice if we could move all the [_handle_remoteconfig](https://github.com/DataDog/dd-trace-py/blob/8bedc18c981af0414424fa350fd030a3d0d2ce10/ddtrace/settings/_config.py#L845-L846) methods out of `ddtrace.setting._core.py` but we can do that in a follow up.